rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/JdbcRecordStorage.java

```diff
@@ -137,18 +174,40 @@ protected EntityStorageRecord readRecord(I id) throws DatabaseException {
      */
     @VisibleForTesting
     @Override
-    protected void writeRecord(I id, EntityStorageRecord record) throws DatabaseException {
+    protected void writeRecord(I id, EntityRecord record) throws DatabaseException {
         checkArgument(record.hasState(), "entity state");
 
         if (containsRecord(id)) {
-            queryFactory.newUpdateEntityQuery(id, record).execute();
+            queryFactory.newUpdateEntityQuery(id, record)
+                        .execute();
         } else {
-            queryFactory.newInsertEntityQuery(id, record).execute();
+            queryFactory.newInsertEntityQuery(id, record)
+                        .execute();
+        }
+    }
+
+    @Override
+    protected void writeRecords(Map<I, EntityRecord> records) {
+        // Map's initial capacity is maximum, meaning no records exist in the storage yet
+        final Map<I, EntityRecord> newRecords = new HashMap<>(records.size());
+
+        for (Map.Entry<I, EntityRecord> unclassifiedRecord : records.entrySet()) {
+            final I id = unclassifiedRecord.getKey();
+            final EntityRecord record = unclassifiedRecord.getValue();
+            if (containsRecord(id)) {
+                queryFactory.newUpdateEntityQuery(id, record)
```

Comment:
@dmdashenkov if the execution never reaches the branch, maybe that's an object decomposition / API design issue?